pkg/hash/hash_test.go

```diff
@@ -28,11 +29,11 @@ func TestHash_WriteAny(t *testing.T) {
 		}
 	}
 
-	X := curve.NewIdentityPoint().ScalarBaseMult(sample.Scalar())
+	X := curve.NewIdentityPoint().ScalarBaseMult(sample.Scalar(rand.Reader))
 	a([]*curve.Point{X, X, X, X})
 
 	a(big.NewInt(35))
-	a(curve.NewIdentityPoint().ScalarBaseMult(sample.Scalar()))
+	a(curve.NewIdentityPoint().ScalarBaseMult(sample.Scalar(rand.Reader)))
 	a([]byte{1, 4, 6})
```

Comment:
If we remove the ability to write point slices then these two cases can be removed